Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,494 @@
+                '50.' => [
+                    'value' => '1',
+                    'wrap' => ',"available":|'
+                ],
+                '60' => 'TEXT',
+                '60.' => [
+                    'stdWrap.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'wrap' => ',"languageUid":"|"'
+                ],
+                '70' => 'USER',
+                '70.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'nav_title',
+                    'stdWrap.' => [
+                        'wrap' => ',"navigationTitle":|'
+                    ]
+                ],
+                '71' => 'USER',
+                '71.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'language',
+                    'stdWrap.' => [
+                        'wrap' => ',"language":|'
+                    ]
+                ],
+                '72' => 'USER',
+                '72.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'locale',
+                    'stdWrap.' => [
+                        'wrap' => ',"locale":|'
+                    ]
+                ],
+                '80' => 'USER',
+                '80.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'hreflang',
+                    'stdWrap.' => [
+                        'wrap' => ',"hreflang":|'
+                    ]
+                ],
+                '81' => 'USER',
+                '81.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'direction',
+                    'stdWrap.' => [
+                        'wrap' => ',"direction":|'
+                    ]
+                ],
+                '99' => 'RESTORE_REGISTER'
+            ]
+        ]
+    ];
+
+    /**
+     * @var array
+     */
+    public $menuDefaults = [
+        'as' => 'languagemenu'
+    ];
+
+    /**
+     * @var string
+     */
+    protected $menuAlternativeSortingField;
```

Comment:
are we able to sort this? is it passed to `sys_language` for sorting?

Reply:
see above, sorting is pending

Reply:
kk

Reply:
done

Reply:
please close change request